tests/codeception/acceptance/AuthenticatedPermissionsCest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * Test the restrictions on authenticated users.
+ */
+class AuthenticatedPermissionsCest {
+
+  /**
+   * Set up a file to test PHP injection.
+   */
+  public function _before(AcceptanceTester $I) {
+    file_put_contents(codecept_data_dir() . '/injection.php', '<?php echo("injection test"); die(); ?>');
+  }
+
+  /**
+   * Remove the php injection file.
+   */
+  public function _after(AcceptanceTester $I) {
+    unlink(codecept_data_dir() . '/injection.php');
```

Comment:
This is SMRT.

Reply:
I cant remember, does `_after` run on both failures and successes?

Reply:
The docs say both `_before` and `_after` get run for every test; it doesn't explicitly say the outcome of the tests matters.  I believe it runs no matter if it's a success or a failure.